Index: app/src/main/java/com/wilderarias/smarta2/ruta/AdaptadorRuta.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/AdaptadorRuta.java	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/java/com/wilderarias/smarta2/ruta/AdaptadorRuta.java	(revision )
@@ -76,7 +76,7 @@
         if (longClickListener != null) {
             longClickListener.onLongClick(v);
         }
-        return false;
+        return true;
     }
 
 
Index: app/src/main/java/com/wilderarias/smarta2/ruta/IngresarVentaActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/IngresarVentaActivity.java	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/java/com/wilderarias/smarta2/ruta/IngresarVentaActivity.java	(revision )
@@ -1,7 +1,9 @@
 package com.wilderarias.smarta2.ruta;
 
+import android.support.v7.app.ActionBar;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.MenuItem;
 
 import com.wilderarias.smarta2.R;
 
@@ -11,5 +13,20 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_ingresar_venta);
+
+        ActionBar actionBar=getSupportActionBar();
+        actionBar.setDisplayHomeAsUpEnabled(true);
+
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
     }
 }
Index: app/src/main/res/layout/activity_ver_productos.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_ver_productos.xml	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/res/layout/activity_ver_productos.xml	(revision )
@@ -86,11 +86,11 @@
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:orientation="horizontal"
-            android:weightSum="4">
+            android:weightSum="7">
 
             <TextView
                 android:layout_width="0dp"
-                android:layout_weight="1"
+                android:layout_weight="2"
                 android:gravity="center"
                 android:textSize="20sp"
                 android:textColor="@color/black"
@@ -98,7 +98,7 @@
                 android:text="@string/codigo"/>
             <TextView
                 android:layout_width="0dp"
-                android:layout_weight="1"
+                android:layout_weight="2"
                 android:gravity="center"
                 android:textSize="20sp"
                 android:textColor="@color/black"
@@ -106,7 +106,7 @@
                 android:text="@string/nombrep"/>
             <TextView
                 android:layout_width="0dp"
-                android:layout_weight="1"
+                android:layout_weight="2"
                 android:gravity="center"
                 android:textSize="20sp"
                 android:textColor="@color/black"
Index: app/src/main/res/layout/activity_ingresar_venta.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_ingresar_venta.xml	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/res/layout/activity_ingresar_venta.xml	(revision )
@@ -9,9 +9,9 @@
     tools:context="com.wilderarias.smarta2.ruta.IngresarVentaActivity">
 
     <TextView
+        android:id="@+id/tposRuta"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="#"
         android:textColor="@color/black"
         android:textStyle="bold"
         android:textSize="30sp"/>
@@ -31,8 +31,7 @@
         <EditText
             android:id="@+id/eCedula"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:hint="1234" />
+            android:layout_height="wrap_content" />
     </LinearLayout>
 
     <LinearLayout
@@ -51,8 +50,7 @@
         <EditText
             android:id="@+id/eNombre"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:hint="Carlos" />
+            android:layout_height="wrap_content" />
     </LinearLayout>
 
     <LinearLayout
@@ -72,7 +70,7 @@
             android:id="@+id/eApellido"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:hint="Restrepo" />
+            />
     </LinearLayout>
 
     <LinearLayout
@@ -92,7 +90,7 @@
             android:id="@+id/eDireccion"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:hint="calle 4" />
+            />
     </LinearLayout>
 
     <LinearLayout
@@ -112,7 +110,7 @@
             android:id="@+id/eTelefono"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:hint="124" />
+          />
     </LinearLayout>
 
     <LinearLayout
@@ -131,8 +129,7 @@
         <EditText
             android:id="@+id/eComentario"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:hint="cerca Iglesia jesuitas" />
+            android:layout_height="wrap_content" />
     </LinearLayout>
 
     <View
Index: app/src/main/res/layout/dialogrutalayout.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/dialogrutalayout.xml	(revision )
+++ app/src/main/res/layout/dialogrutalayout.xml	(revision )
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
+
+    <LinearLayout
+        android:padding="25dp"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="5dp"
+        android:orientation="horizontal">
+
+        <ImageView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:src="@drawable/ic_agregar_abono" />
+
+        <TextView
+            android:id="@+id/tIngresarAbono"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_marginLeft="10dp"
+            android:gravity="center_vertical"
+            android:textSize="20sp"
+            android:textColor="@color/black"
+            android:text="@string/ingresarabono" />
+    </LinearLayout>
+
+    <View
+        android:layout_width="match_parent"
+        android:layout_height="2dp"
+        android:background="@color/black"></View>
+
+    <LinearLayout
+        android:padding="25dp"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="5dp"
+        android:orientation="horizontal">
+
+        <ImageView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:src="@drawable/ic_agregar_abono" />
+
+        <TextView
+            android:id="@+id/tNuevaVenta"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_marginLeft="10dp"
+            android:textColor="@color/black"
+            android:textSize="20sp"
+            android:gravity="center_vertical"
+            android:text="@string/nuevoprestamo" />
+    </LinearLayout>
+</LinearLayout>
\ No newline at end of file
Index: app/src/main/java/com/wilderarias/smarta2/ruta/AdaptadorProductos.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/AdaptadorProductos.java	(revision )
+++ app/src/main/java/com/wilderarias/smarta2/ruta/AdaptadorProductos.java	(revision )
@@ -0,0 +1,50 @@
+package com.wilderarias.smarta2.ruta;
+
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+import com.wilderarias.smarta2.R;
+
+import org.w3c.dom.Text;
+
+import java.util.ArrayList;
+
+/**
+ * Created by WilderArias on 11/16/2017.
+ */
+
+public class AdaptadorProductos extends ArrayAdapter {
+
+    ArrayList<ProductosData> productosData;
+
+    public AdaptadorProductos(Context context, ArrayList<ProductosData> productosData){
+        super(context, R.layout.productos_list,productosData);
+        this.productosData=productosData;
+    }
+
+    @NonNull
+    @Override
+    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+        LayoutInflater inflater = LayoutInflater.from(getContext());
+        View item = inflater.inflate(R.layout.productos_list, null);
+
+        TextView tIdProd=item.findViewById(R.id.tIdProd);
+        TextView tNombreProd=item.findViewById(R.id.tNombreProd);
+        TextView tCantidadProd=item.findViewById(R.id.tCantidadProd);
+        TextView tValorProd=item.findViewById(R.id.tValorProd);
+
+        tIdProd.setText(productosData.get(position).getCodigoP()+"-"+String.valueOf(productosData.get(position).getCodigoAP()));
+        tNombreProd.setText(productosData.get(position).getNombreP());
+        tCantidadProd.setText(String.valueOf(productosData.get(position).getCantidadP()));
+        tValorProd.setText(String.valueOf(productosData.get(position).getValorNeto()));
+
+        return item;
+    }
+}
Index: app/src/main/java/com/wilderarias/smarta2/ruta/DetalleFacturaActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/DetalleFacturaActivity.java	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/java/com/wilderarias/smarta2/ruta/DetalleFacturaActivity.java	(revision )
@@ -117,19 +117,31 @@
     public boolean onOptionsItemSelected(MenuItem item) {
 
         Intent intent;
+        String fecha = "" + rutaData.getDiaRegistroV() + "/" + rutaData.getMesRegistroV() + "/" + rutaData.getAnoRegistroV();
+
         switch (item.getItemId()) {
             case R.id.actCliente:
                 intent = new Intent(DetalleFacturaActivity.this, ActualizarClienteActivity.class);
                 intent.putExtra("idCliente", rutaData.getIdCliente());
+                intent.putExtra("direccion",rutaData.getDireccionC());
+                intent.putExtra("telefono",rutaData.getTelefonoC());
+                intent.putExtra("comentario",rutaData.getComentarioC());
                 startActivity(intent);
                 return true;
             case R.id.verProductos:
                 intent = new Intent(DetalleFacturaActivity.this, VerProductosActivity.class);
                 intent.putExtra("idFactura", rutaData.getIdFacturaVenta());
+                intent.putExtra("nombreC", rutaData.getNombreC() + " " + rutaData.getApellidoC());
+                intent.putExtra("fechaFactura", fecha);
+                intent.putExtra("valorV", rutaData.getValorVenta());
                 startActivity(intent);
                 return true;
             case R.id.verAbononos:
                 intent = new Intent(DetalleFacturaActivity.this, VerAbonosActivity.class);
+                intent.putExtra("nombreC", rutaData.getNombreC() + " " + rutaData.getApellidoC());
+                intent.putExtra("fechaFactura", fecha);
+                intent.putExtra("valorV", rutaData.getValorVenta());
+                intent.putExtra("saldoV", rutaData.getSaldoCredito());
                 intent.putExtra("idFactura", rutaData.getIdFacturaVenta());
                 startActivity(intent);
                 return true;
Index: app/src/main/res/layout/productos_list.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/productos_list.xml	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/res/layout/productos_list.xml	(revision )
@@ -7,7 +7,7 @@
     <TextView
         android:id="@+id/tIdProd"
         android:layout_width="0dp"
-        android:layout_weight="1"
+        android:layout_weight="2"
         android:layout_height="wrap_content"
         android:textColor="@color/black"
         android:textSize="25sp" />
@@ -18,7 +18,7 @@
         android:layout_weight="2"
         android:layout_height="wrap_content"
         android:textColor="@color/black"
-        android:textSize="25sp" />
+        android:textSize="25sp"/>
 
     <TextView
         android:id="@+id/tCantidadProd"
@@ -26,14 +26,16 @@
         android:layout_weight="1"
         android:layout_height="wrap_content"
         android:textColor="@color/black"
-        android:textSize="25sp" />
+        android:textSize="25sp"
+        android:gravity="center"/>
 
     <TextView
         android:id="@+id/tValorProd"
         android:layout_width="0dp"
-        android:layout_weight="2"
+        android:layout_weight="1"
         android:layout_height="wrap_content"
         android:textColor="@color/black"
-        android:textSize="25sp" />
+        android:textSize="25sp"
+        android:gravity="center"/>
 
 </LinearLayout>
\ No newline at end of file
Index: app/src/main/java/com/wilderarias/smarta2/ruta/RutaFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/RutaFragment.java	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/java/com/wilderarias/smarta2/ruta/RutaFragment.java	(revision )
@@ -33,6 +33,7 @@
     private ArrayList<RutaData> data = new ArrayList<RutaData>();
     private FirebaseDatabase firebaseDatabase;
     private DatabaseReference myRef, myRef2;
+    AdaptadorRuta rutaAdaptadorRuta;
 
     public RutaFragment() {
         // Required empty public constructor
@@ -42,13 +43,14 @@
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
 
+        Log.i("onCreateView","rutaFragament");
         View view = inflater.inflate(R.layout.ruta_fragment_ruta, container, false);
         recyclerView = view.findViewById(R.id.listRuta);
 
         firebaseDatabase = FirebaseDatabase.getInstance();
         myRef = firebaseDatabase.getReference().child("venta");
         myRef2 = firebaseDatabase.getReference().child("cliente");
-        final AdaptadorRuta rutaAdaptadorRuta = new AdaptadorRuta(data);
+        rutaAdaptadorRuta = new AdaptadorRuta(data);
 
         recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext(), LinearLayoutManager.VERTICAL, false));
         recyclerView.setAdapter(rutaAdaptadorRuta);
@@ -78,7 +80,7 @@
                             data.get(i).setApellidoC(clienteSnapshot.child("apellidoC").getValue().toString());
                             data.get(i).setComentarioC(clienteSnapshot.child("comentarioC").getValue().toString());
                             data.get(i).setDireccionC(clienteSnapshot.child("direccionC").getValue().toString());
-                            data.get(i).setTelefonoC((long)clienteSnapshot.child("telefonoC").getValue());
+                            data.get(i).setTelefonoC((long) clienteSnapshot.child("telefonoC").getValue());
                         }
                     }
                 }
@@ -100,25 +102,42 @@
         rutaAdaptadorRuta.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                //Toast.makeText(getContext(), "click sobre" + recyclerView.getChildPosition(v), Toast.LENGTH_SHORT).show();
+
+                Log.i("click", "YES");
                 Intent intent = new Intent(getContext(), DetalleFacturaActivity.class);
-                Gson gson=new Gson();
-                String gsonData=gson.toJson(data.get(recyclerView.getChildPosition(v)));
-               // intent.putExtra("idFactura",data.get(recyclerView.getChildPosition(v)).getIdFacturaVenta());
-                intent.putExtra("data",gsonData);
+                Gson gson = new Gson();
+                String gsonData = gson.toJson(data.get(recyclerView.getChildPosition(v)));
+                // intent.putExtra("idFactura",data.get(recyclerView.getChildPosition(v)).getIdFacturaVenta());
+                intent.putExtra("data", gsonData);
                 startActivity(intent);
+
             }
         });
 
         rutaAdaptadorRuta.setOnLongClickListener(new View.OnLongClickListener() {
             @Override
             public boolean onLongClick(View v) {
-                Toast.makeText(getContext(), "LongClick sobre" + recyclerView.getChildPosition(v), Toast.LENGTH_SHORT).show();
+                // Toast.makeText(getContext(), "LongClick sobre" + recyclerView.getChildPosition(v), Toast.LENGTH_SHORT).show();
+                OpcionesRutaDialogFragment dialog = new OpcionesRutaDialogFragment();
+                dialog.show(getFragmentManager(), "opcionesruta");
                 return true;
             }
         });
         return view;
     }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        Log.i("onResume","rutaFragament");
+
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        Log.i("onStop","rutaFragament");
+    }
 }
 
 
Index: app/src/main/java/com/wilderarias/smarta2/ruta/ProductosData.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/ProductosData.java	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/java/com/wilderarias/smarta2/ruta/ProductosData.java	(revision )
@@ -6,18 +6,18 @@
 
 public class ProductosData {
     private String codigoP,nombreP;
-    private long cantidadP,codigoAP,idFacturaVenta,idSucursal,numerP,valorNeto;
+    private long cantidadP,codigoAP,idFacturaVenta,idSucursal,numeroP,valorNeto;
 
     public ProductosData() {
     }
 
-    public ProductosData(String codigoP, long cantidadP, long codigoAP, long idFacturaVenta, long idSucursal, long numerP) {
+    public ProductosData(String codigoP, long cantidadP, long codigoAP, long idFacturaVenta, long idSucursal, long numeroP) {
         this.codigoP = codigoP;
         this.cantidadP = cantidadP;
         this.codigoAP = codigoAP;
         this.idFacturaVenta = idFacturaVenta;
         this.idSucursal = idSucursal;
-        this.numerP = numerP;
+        this.numeroP = numeroP;
     }
 
     public String getNombreP() {
@@ -76,11 +76,11 @@
         this.idSucursal = idSucursal;
     }
 
-    public long getNumerP() {
-        return numerP;
+    public long getNumeroP() {
+        return numeroP;
     }
 
-    public void setNumerP(long numerP) {
-        this.numerP = numerP;
+    public void setNumeroP(long numeroP) {
+        this.numeroP = numeroP;
     }
 }
Index: app/src/main/res/layout/activity_actualizar_cliente.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_actualizar_cliente.xml	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/res/layout/activity_actualizar_cliente.xml	(revision )
@@ -24,7 +24,7 @@
             android:id="@+id/eDireccion"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:hint="calle 4" />
+            />
     </LinearLayout>
 
     <LinearLayout
@@ -43,8 +43,7 @@
         <EditText
             android:id="@+id/eTelefono"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:hint="1234" />
+            android:layout_height="wrap_content" />
     </LinearLayout>
 
     <LinearLayout
@@ -64,7 +63,7 @@
             android:id="@+id/eComentario"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:hint="Informacion adicional cliente" />
+            />
     </LinearLayout>
 
     <LinearLayout
@@ -75,20 +74,24 @@
         android:weightSum="2">
 
         <Button
+            android:id="@+id/bAceptar"
             android:layout_width="0dp"
             android:layout_height="wrap_content"
             android:layout_marginRight="5dp"
             android:layout_weight="1"
             android:background="@color/colorAccent"
             android:text="@string/aceptar"
-            android:textSize="20sp" />
+            android:textSize="20sp"
+            android:onClick="click"/>
 
         <Button
+            android:id="@+id/bCancelar"
             android:layout_width="0dp"
             android:layout_height="wrap_content"
             android:layout_weight="1"
             android:background="@color/colorAccent"
             android:text="@string/cancelar"
-            android:textSize="20sp" />
+            android:textSize="20sp"
+            android:onClick="click"/>
     </LinearLayout>
 </LinearLayout>
Index: app/src/main/res/values/arrays.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/arrays.xml	(revision )
+++ app/src/main/res/values/arrays.xml	(revision )
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string-array name="opc_ruta_array" >
+        <item>@string/ingresarabono</item>
+        <item>@string/nuevoprestamo</item>
+    </string-array>
+</resources>
\ No newline at end of file
Index: app/src/main/java/com/wilderarias/smarta2/ruta/ActualizarClienteActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/ActualizarClienteActivity.java	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/java/com/wilderarias/smarta2/ruta/ActualizarClienteActivity.java	(revision )
@@ -1,13 +1,107 @@
 package com.wilderarias.smarta2.ruta;
 
+import android.app.TaskStackBuilder;
+import android.support.v7.app.ActionBar;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.text.SpannableStringBuilder;
+import android.util.Log;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
 import com.wilderarias.smarta2.R;
+
+import org.w3c.dom.Text;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
 public class ActualizarClienteActivity extends AppCompatActivity {
+    private String direccionC,comentario,idCliente;
+    private long telefonoC;
+    private EditText eDireccionC,eTelefonoC,eComentario;
+    private String direccion,comentarioc;
+    private long telefono;
+    FirebaseDatabase firebaseDatabase=FirebaseDatabase.getInstance();
+    DatabaseReference myRef;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_actualizar_cliente);
+
+        ActionBar actionBar=getSupportActionBar();
+        actionBar.setDisplayHomeAsUpEnabled(true);
+
+        if(getIntent()!=null){
+            idCliente=getIntent().getStringExtra("idCliente");
+            direccionC=getIntent().getStringExtra("direccion");
+            telefonoC=getIntent().getLongExtra("telefono",0);
+            comentario=getIntent().getStringExtra("comentario");
+        }
+        eDireccionC=findViewById(R.id.eDireccion);
+        eTelefonoC=findViewById(R.id.eTelefono);
+        eComentario=findViewById(R.id.eComentario);
+
+        eDireccionC.setText(direccionC);
+        eTelefonoC.setText(String.valueOf(telefonoC));
+        eComentario.setText(comentario);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()){
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    public void click(View view) {
+        int bOpc=view.getId();
+        switch (bOpc){
+            case R.id.bAceptar:
+                direccion=eDireccionC.getText().toString();
+                telefono=Long.parseLong(eTelefonoC.getText().toString()) ;
+                comentarioc=eComentario.getText().toString();
+
+                myRef=firebaseDatabase.getReference("cliente");
+                myRef.addValueEventListener(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+                        for (DataSnapshot actSnapshot:dataSnapshot.getChildren()){
+                            if (Objects.equals(actSnapshot.child("idCliente").getValue().toString(),idCliente)){
+                                Map<String,Object> newData = new HashMap<>();
+                                newData.put("direccionC",direccion);
+                                newData.put("telefonoC",telefono);
+                                newData.put("comentarioC",comentario);
+                                Log.i("ActualizarCliente","actualizando");
+                                myRef.child(actSnapshot.getKey()).updateChildren(newData);
+                                break;
+                            }
+                        }
+                    }
+
+                    @Override
+                    public void onCancelled(DatabaseError databaseError) {
+
+                    }
+                });
+
+                break;
+            case R.id.bCancelar:
+                onBackPressed();
+                break;
+        }
     }
 }
Index: app/src/main/java/com/wilderarias/smarta2/ruta/OpcionesRutaDialogFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/OpcionesRutaDialogFragment.java	(revision )
+++ app/src/main/java/com/wilderarias/smarta2/ruta/OpcionesRutaDialogFragment.java	(revision )
@@ -0,0 +1,70 @@
+package com.wilderarias.smarta2.ruta;
+
+import android.app.Dialog;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.support.v4.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.wilderarias.smarta2.R;
+
+/**
+ * Created by WilderArias on 11/17/2017.
+ */
+
+public class OpcionesRutaDialogFragment extends DialogFragment {
+
+    public OpcionesRutaDialogFragment() {
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        AlertDialog.Builder builder=new AlertDialog.Builder(getActivity());
+        LayoutInflater inflater=getActivity().getLayoutInflater();
+        View v=inflater.inflate(R.layout.dialogrutalayout,null);
+        builder.setView(v);
+
+        TextView tIngresarAbono=v.findViewById(R.id.tIngresarAbono);
+        TextView tNuevaVenta=v.findViewById(R.id.tNuevaVenta);
+
+        tIngresarAbono.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View v) {
+
+                dismiss();
+            }
+        });
+
+        tNuevaVenta.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent=new Intent(getContext(),IngresarVentaActivity.class);
+                startActivity(intent);
+                dismiss();
+           }
+        });
+
+        return builder.create();
+    }
+}
+
+/*
+
+  AlertDialog.Builder builder=new AlertDialog.Builder(getActivity());
+               builder.setItems(R.array.opc_ruta_array, new DialogInterface.OnClickListener() {
+                @Override
+            public void onClick(DialogInterface dialog, int which) {
+                Toast.makeText(getActivity(),"selecciono"+which,Toast.LENGTH_SHORT).show();
+            }
+        });
+
+        return builder.create();
+
+ */
\ No newline at end of file
Index: app/src/main/java/com/wilderarias/smarta2/ruta/VerProductosActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/wilderarias/smarta2/ruta/VerProductosActivity.java	(revision 17cbcaf173946c087bc39ef2425ed7987f2b7235)
+++ app/src/main/java/com/wilderarias/smarta2/ruta/VerProductosActivity.java	(revision )
@@ -17,6 +17,8 @@
 
 import java.util.ArrayList;
 import java.util.Objects;
+import java.util.Timer;
+import java.util.TimerTask;
 
 public class VerProductosActivity extends AppCompatActivity {
 
@@ -55,10 +57,15 @@
         tValorVenta.setText(String.valueOf(valorVenta));
         tFechaFactura.setText(fechaFactura);
 
+        final AdaptadorProductos adaptadorProductos=new AdaptadorProductos(getApplicationContext(),productosData);
+
+        lProd.setAdapter(adaptadorProductos);
+
         firebaseDatabase=FirebaseDatabase.getInstance();
         myRef=firebaseDatabase.getReference().child("productos_venta");
         myRef2=firebaseDatabase.getReference().child("producto");
         myRef3=firebaseDatabase.getReference().child("articulo_producto");
+        Timer timer=new Timer();
 
         myRef.addValueEventListener(new ValueEventListener() {
             @Override
@@ -77,44 +84,53 @@
             }
         });
 
-        myRef2.addValueEventListener(new ValueEventListener() {
-            @Override
-            public void onDataChange(DataSnapshot dataSnapshot) {
-                for (int i=0;i<productosData.size();i++){
-                    for (DataSnapshot nomProSnapshot:dataSnapshot.getChildren() ){
-                        Log.i("VerProductosActivity","myRef2");
-                        if (Objects.equals(nomProSnapshot.child("codigoP").getValue().toString(),productosData.get(i).getCodigoP())){
-                            productosData.get(i).setNombreP(nomProSnapshot.child("nombreP").getValue().toString());
-                        }
-                    }
-                }
-            }
+
+        TimerTask timerTask=new TimerTask() {
+            @Override
+            public void run() {
+
+                myRef2.addValueEventListener(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+                        for (int i=0;i<productosData.size();i++){
+                            for (DataSnapshot nomProSnapshot:dataSnapshot.getChildren() ){
+                                Log.i("VerProductosActivity","myRef2");
+                                if (Objects.equals(nomProSnapshot.child("codigoP").getValue().toString(),productosData.get(i).getCodigoP())){
+                                    productosData.get(i).setNombreP(nomProSnapshot.child("nombreP").getValue().toString());
+                                }
+                            }
+                        }
+                    }
 
-            @Override
-            public void onCancelled(DatabaseError databaseError) {
+                    @Override
+                    public void onCancelled(DatabaseError databaseError) {
 
-            }
-        });
+                    }
+                });
 
-        myRef3.addValueEventListener(new ValueEventListener() {
-            @Override
-            public void onDataChange(DataSnapshot dataSnapshot) {
-                for (int j=0;j<productosData.size();j++){
-                    for (DataSnapshot valProSnapshot:dataSnapshot.getChildren() ){
-                        Log.i("VerProductosActivity","myRef3");
-                        if (Objects.equals(valProSnapshot.child("codigoP").getValue().toString(),productosData.get(j).getCodigoP())){
-                            productosData.get(j).setValorNeto((long)valProSnapshot.child("valorNeto").getValue());
-                        }
-                    }
-                }
-            }
+                myRef3.addValueEventListener(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+                        for (int j=0;j<productosData.size();j++){
+                            for (DataSnapshot valProSnapshot:dataSnapshot.getChildren() ){
+                                Log.i("VerProductosActivity","myRef3");
+                                if (Objects.equals(valProSnapshot.child("codigoP").getValue().toString(),productosData.get(j).getCodigoP())){
+                                    productosData.get(j).setValorNeto((long)valProSnapshot.child("valorNeto").getValue());
+                                }
+                            }
+                        }
+                        adaptadorProductos.notifyDataSetChanged();
+                    }
 
-            @Override
-            public void onCancelled(DatabaseError databaseError) {
+                    @Override
+                    public void onCancelled(DatabaseError databaseError) {
 
-            }
-        });
+                    }
+                });
 
+            }
+        };
+        timer.schedule(timerTask,400);
     }
 
     @Override
